addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -3215,11 +3231,53 @@ options.registry.MegaMenuLayout = options.registry.SelectTemplate.extend({
     _computeWidgetState: function (methodName, params) {
         if (methodName === 'selectTemplate') {
             const templateDefiningClass = this.containerEl.querySelector('section')
-                .classList.value.split(' ').filter(cl => cl.startsWith('s_mega_menu'))[0];
+                .classList.value.split(' ').filter(cl => cl.startsWith('s_mega_menu_'))[0];
             return `website.${templateDefiningClass}`;
         }
         return this._super(...arguments);
     },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Puts back original template.
+     *
+     * @private
+     * @param {OdooEvent} ev
+     */
+    async _onResetTemplate(ev) {
+        const widgetValue = this._computeWidgetState('selectTemplate');
+        const template = await this._rpc({
+            model: 'ir.ui.view',
+            method: 'render_public_asset',
+            args: [widgetValue, {}],
+        });
+        this.containerEl.insertAdjacentHTML('beforeend', template);
+    },
+});
+
+/**
+ * Hides delete button.
+ */
+options.registry.NoDeleteButton = options.Class.extend({
+    forceNoDeleteButton: true,
```

Comment:
Why not setting this on the already existing option targeting the mega menu?

Reply:
The existing option targets the Mega Menu and not the block below we actually want to target. 
![megamenuselector](https://user-images.githubusercontent.com/49784121/133091154-c66129f5-7935-4670-b989-26f601848067.png)

Reply:
Yep I saw that, forgot to remove that comment before sending the review. However, do not create an option named `NoDeleteButton`, that kinda makes wondering why there is a forceNoDeleteButton option per option if there is a dedicated Option Class to do that. Name is specifically something like "MegaMenuNoDelete".